pkg/scheduler/scheduler_test.go

```diff
@@ -483,8 +484,8 @@ func TestSchedulerQuerierMetrics(t *testing.T) {
 		return err == nil
 	}, time.Second, 10*time.Millisecond, "expected cortex_query_scheduler_connected_querier_clients metric to be incremented after querier connected")
 
-	require.NoError(t, querierLoop.CloseSend())
 	cancel()
+	require.NoError(t, util.CloseAndExhaust[*schedulerpb.SchedulerToQuerier](querierLoop))
```

Comment:
Note to reviewers: moving the call to `cancel()` before closing the stream is required because otherwise `CloseAndExhaust()` will block: the stream has not ended, and the server does not know to close the stream without the cancellation. (Previously, `CloseSend()` returned immediately, and the call to `cancel()` was what made the test pass.)